vendor/engines/split_accounts/app/services/split_accounts/split_order_detail_simulator.rb

```diff
@@ -0,0 +1,90 @@
+module SplitAccounts
+
+  # TODO: maybe refactor some of this into SplitSimulator
+  # TODO: we can refactor this code to fewer lines and iterate fewer times
+  class SplitOrderDetailSimulator
+
+    SimulatedSplit = Struct.new(
+      :split,
+      :quantity,
+      :actual_cost,
+      :actual_subsidy,
+      :estimated_cost,
+      :estimated_subsidy
+    )
+
+    attr_accessor :order_detail, :account, :splits
+
+    def initialize(order_detail)
+      @order_detail = order_detail
+      @account = order_detail.account
+      @splits = account.splits
+    end
+
+    def simulated_order_details
+      simulated_splits.map do |item|
+        simulated_order_detail(item)
+      end
+    end
+
+    def simulated_order_detail(item)
+      clone = SplitOrderDetailDecorator.new(order_detail.deep_clone)
+
+      clone.account = item.split.subaccount
+      clone.quantity = item.quantity
+      clone.actual_cost = item.actual_cost
+      clone.actual_subsidy = item.actual_subsidy
+      clone.estimated_cost = item.estimated_cost
+      clone.estimated_subsidy = item.estimated_subsidy
+
+      clone
+    end
+
+    def simulated_splits
+      items = splits.map { |split| simulated_split(split) }
```

Comment:
As we get deeper into these methods, it becomes very difficult to remember what `items` is. I'm also having trouble wrapping my head around `SimulatedSplit` and how it's different from the simulated order detail.